gui/default/syncthing/core/syncthingController.js

```diff
@@ -549,10 +567,28 @@ angular.module('syncthing.core')
                 $scope.connectionsTotal = data.total;
 
                 data = data.connections;
+                var currentAddresses = [];
                 for (id in data) {
                     if (!data.hasOwnProperty(id)) {
                         continue;
                     }
+                    if (!(id in $scope.devices)) {
+                        // Avoid errors when called before first updateLocalConfig()
+                        continue;
+                    }
+                    if (!(id in $scope.idToRemoteGUI)) {
+                        $scope.idToRemoteGUI[id] = "";
+                    }
+                    var port = $scope.devices[id].remoteGUIPort;
+                    var isNotRelayConnection = !data[id].type.includes("relay");
+                    if ($scope.showRemoteGUI && data[id].address != "" && isNotRelayConnection && port > 0) {
+                        var newAddress = "http://" + replaceAddressPort(data[id].address, port);
+                        currentAddresses.push(newAddress);
+                        if (!(newAddress in $scope.remoteGUICache)) {
```

Comment:
While it won't work, it's possible that two devices behind the same NAT, thus same IP, and are both configured with 8384. Then the result of the existing lookup should be applied to `idToRemoteGUI`.

Reply:
Good point.  Not sure how to best implement that, especially after trying to get rid of the cached success value.

Reply:
If the ip address could be the same, can we concatenate a portion of the device id into the ip address? Of course, we would have to parse it in the appropriate locations.

Reply:
Do you mean to add that in the cache variable? I don't see a real benefit there. Reusing the same probing result is the right thing to do, after all it would be a misconfiguration to expect different GUIs under the same port for devices behind the same NAT IP. We cannot remedy that, but at least be fair and display both while still avoiding repeated probing on the same URL.

I have a draft already, just need to clean up, test and push when my schedule permits.